src/test/ui/generic-associated-types/bugs/issue-88382.stderr

```diff
@@ -2,18 +2,18 @@ error[E0631]: type mismatch in function arguments
   --> $DIR/issue-88382.rs:27:40
    |
 LL |     do_something(SomeImplementation(), test);
-   |     ------------                       ^^^^ expected signature of `for<'a> fn(&mut <SomeImplementation as Iterable>::Iterator<'a>) -> _`
+   |     ------------                       ^^^^ expected signature of `for<'r> fn(&'r mut std::iter::Empty<usize>) -> _`
```

Comment:
This test highlights that something wonky is going on. It changes were we aren't able to normalize.

Reply:
Yeah, this is very interesting. Possibly due to changes in the ordering of resolving the inference variable in `&mut <_ as Iterable>::Iterator<'a>` and normalizing it. Now that we're eagerly normalizing the trait ref, we probably are substituting that value before we have a chance to unify it with something else, possibly another projection type with an inference variable.